File: daily_problems/2025/07/0702/personal_submission/cf358c_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/all"

using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

// Let's set a bit and flow!
// I came, I divided, I conquered!
// using namespace __gnu_cxx;
// using namespace __gnu_pbds;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int q;
    cin >> q;

    vector<int> cur;
    int v1 = 0, v2 = 0, v3 = 0;

    while (q --) {
        int x;
        cin >> x;

        if (x) {
            if (x > v1) {
                v3 = x;
                swap(v2, v3);
                swap(v1, v2);
            }
            else if (x > v2) {
                v3 = x;
                swap(v2, v3);
            }
            else if (x > v3)
                v3 = x;
            cur.emplace_back(x);
        }
        else {
            vector<string> tmp;

            for (auto &v: cur) {
                if (v == v1) {
                    cout << "pushStack\n";
                    tmp.emplace_back("popStack");
                    v1 = 0;
                }
                else if (v == v2) {
                    cout << "pushQueue\n";
                    tmp.emplace_back("popQueue");
                    v2 = 0;
                }
                else if (v == v3) {
                    cout << "pushFront\n";
                    tmp.emplace_back("popFront");
                    v3 = 0;
                }
                else cout << "pushBack\n";
            }
            cur.clear();

            if (tmp.size()) {
                cout << tmp.size();
                for (auto &x: tmp)
                    cout << ' ' << x;
            }
            else cout << 0;
            cout << '\n';
        }
    }

    for (auto &v: cur) cout << "pushStack\n";

    return 0;
}

File: daily_problems/2025/07/0702/personal_submission/cf358c_yawn_sean.py
# Submission link: https://codeforces.com/contest/358/submission/327016350
def main():
    q = II()
    cur = []
    outs = []

    v1, v2, v3 = 0, 0, 0

    for _ in range(q):
        x = II()
        
        if x:
            if x > v1:
                v1, v2, v3 = x, v1, v2
            elif x > v2:
                v2, v3 = x, v2
            elif x > v3:
                v3 = x
            cur.append(x)
        else:
            tmp = []
            for v in cur:
                if v == v1:
                    outs.append('pushStack')
                    tmp.append('popStack')
                    v1 = 0
                elif v == v2:
                    outs.append('pushQueue')
                    tmp.append('popQueue')
                    v2 = 0
                elif v == v3:
                    outs.append('pushFront')
                    tmp.append('popFront')
                    v3 = 0
                else:
                    outs.append('pushBack')
            
            if tmp: outs.append(f"{len(tmp)} {' '.join(tmp)}")
            else: outs.append('0')
            cur.clear()

    for v in cur:
        outs.append('pushStack')

    print('\n'.join(outs))